src/openforms/submissions/api/viewsets.py

```diff
@@ -233,13 +232,8 @@ def _complete(self, request, *args, **kwargs):
         ---
         """
         submission = self.get_object()
-        validation_serializer = validate_submission_completion(
-            submission, request=request
-        )
-        if validation_serializer is not None:
-            return Response(
-                validation_serializer.data, status=status.HTTP_400_BAD_REQUEST
-            )
+
+        validate_submission_completion(submission, request=request)
```

Comment:
looks good, clever trick with the `MaxLengthValidator` :smile: 

one thing I would change is make this function return the serializer and call the `serializer.is_valid(raise_exception=True)` and `serializer.save()` in the view, rather than lumping that into the validation function (so it also makes sense to rename the function to `get_completion_validation_serializer` to have a more accurate name).